src/NLog/Targets/FileTarget.cs

```diff
@@ -237,8 +237,11 @@ public Layout FileName
 
                 fileName = value;
 
-                RefreshFileArchive();
-                RefreshArchiveFilePatternToWatch();
+                if (IsInitialized)
+                {
+                    RefreshFileArchive();
+                    RefreshArchiveFilePatternToWatch();
```

Comment:
These two methods are called from a number of other properties as well. Perhaps we should move the `IsInitialized` check into the `RefreshFileArchive()` and `RefreshArchiveFilePatternToWatch()` methods.